src/snovault/elasticsearch/indexer.py

```diff
@@ -16,26 +16,282 @@
 from urllib3.exceptions import ReadTimeoutError
 from .interfaces import (
     ELASTIC_SEARCH,
-    INDEXER,
+    INDEXER
 )
+from snovault import CONNECTION
 import datetime
 import logging
 import pytz
 import time
 import copy
 
-
+es_logger = logging.getLogger("elasticsearch")
+es_logger.setLevel(logging.ERROR)
 log = logging.getLogger(__name__)
 SEARCH_MAX = 99999  # OutOfMemoryError if too high
 
-
 def includeme(config):
     config.add_route('index', '/index')
     config.scan(__name__)
     registry = config.registry
     registry[INDEXER] = Indexer(registry)
 
 
+class IndexerState(object):
+    # Keeps track of uuids and indexer state by cycle.  Also handles handoff of uuids to followup indexer
+    def __init__(self, es, key, title='primary'):
+        self.es = es
+        self.key = key  # "indexerstate"
+        self.title           = title
+        self.state_id        = self.title + '_indexer'       # State of the current or last cycle
+        self.todo_set        = self.title + '_in_progress'   # one cycle of uuids, sent to the Secondary Indexer
+        #self.failed_set      = self.title + '_failed'
```

Comment:
I guess these were for the defunct redis version?  Probably should just drop them.

Reply:
This state object IS used and most of its methods are used in indexeer.py.  Those not are used in secondary_indexer.py.